web-ui/src/streaming/import/InsertionTable.tsx

```diff
@@ -0,0 +1,156 @@
+    }
+  }, [pipelineRevisionQuery.isLoading, pipelineRevisionQuery.isError, pipelineRevisionQuery.data, name])
+
+  return relation ? (
+    <Card>
+      <DataGridPro
+        apiRef={apiRef}
+        autoHeight
+        editMode='cell'
+        density='compact'
+        loading={isLoading}
+        rows={rows}
+        disableColumnFilter
+        initialState={{
+          columns: {
+            columnVisibilityModel: {
+              genId: false
+            }
+          }
+        }}
+        getRowId={(row: Row) => row.genId}
+        columns={relation.fields
+          .map((col: Field) => {
+            return {
+              field: col.name,
+              headerName: col.name,
+              description: col.name,
+              flex: 1,
+              editable: true,
+              type: sqlTypeToDataGridType(col),
+              preProcessEditCellProps: (params: GridPreProcessEditCellProps) => {
+                // We'll add support for this once we have JSON format + better
+                // errors from backend
+                const hasError = false
+                return { ...params.props, error: hasError }
+              },
+              valueGetter: (params: GridValueGetterParams) => params.row.record[col.name],
+              valueFormatter: (params: GridValueFormatterParams<any>) => {
+                return getValueFormatter(col.columntype)(params.value)
+              },
+              valueParser: (value: any) => {
+                // It looks like this doesn't do anything -- it just returns the
+                // value, but I found without having defined this, if you edit a
+                // cell and enter e.g., 'ab' into a number field it will throw
+                // an exception, whereas with this it will do the 'right' thing
+                // and set it to an empty string...
+                return value
+              },
+              valueSetter: (params: GridValueSetterParams) => {
+                const row = params.row
+                row.record[col.name] = params.value
+                return row
+              }
+            }
+          })
+          .concat([
+            {
+              field: 'genId',
+              headerName: 'genId',
+              description: 'Index relative to the current set of rows we are adding.',
+              flex: 0.5,
+              editable: false,
+              type: 'number',
+              preProcessEditCellProps: (params: GridPreProcessEditCellProps) => {
+                const hasError = false
+                return { ...params.props, error: hasError }
+              },
+              valueFormatter: (params: GridValueFormatterParams<any>) => {
+                return params.value
+              },
+              valueParser: (value: any) => {
+                return value
+              },
+              valueGetter: (params: any) => params.row.genId,
+              valueSetter: (params: GridValueSetterParams) => {
+                return params.row
```

Comment:
why is this a setter?

Reply:
we have the setter for the other columns, so typescript complained about incompatible types without it